procrastinate/sql/migrations/03.00.00_50_post_cancel_notification.sql

```diff
@@ -1,66 +1,103 @@
--- Add an 'abort_requested' column to the procrastinate_jobs table
-ALTER TABLE procrastinate_jobs ADD COLUMN abort_requested boolean DEFAULT false NOT NULL;
+-- These are old versions of function, that we needed
+-- to keep around for backwards compatibility. We can now safely drop them.
+DROP FUNCTION IF EXISTS procrastinate_finish_job(
+    integer,
+    procrastinate_job_status,
+    timestamp with time zone,
+    boolean
+);
+DROP FUNCTION IF EXISTS procrastinate_defer_job(
+    character varying,
+    character varying,
+    text,
+    text,
+    jsonb,
+    timestamp with time zone
+);
+DROP FUNCTION IF EXISTS procrastinate_defer_periodic_job(
+    character varying,
+    character varying,
+    character varying,
+    character varying,
+    character varying,
+    bigint,
+    jsonb
+);
+DROP FUNCTION IF EXISTS procrastinate_retry_job(
+    bigint,
+    timestamp with time zone
+);
+DROP FUNCTION IF EXISTS procrastinate_retry_job(
+    bigint,
+    timestamp with time zone,
+    integer,
+    character varying,
+    character varying
+);
 
--- Set abort requested flag on all jobs with 'aborting' status
+-- Remove all traces of the "aborting" status
+-- Last sanity update in case the trigger didn't work 100% of the time
 UPDATE procrastinate_jobs SET abort_requested = true WHERE status = 'aborting';
 
+DROP TRIGGER IF EXISTS procrastinate_trigger_sync_abort_requested_with_status_v1 ON procrastinate_jobs;
+DROP FUNCTION IF EXISTS procrastinate_sync_abort_requested_with_status_v1;
+
+-- Create a new enum type without 'aborting'
+CREATE TYPE procrastinate_job_status_v1 AS ENUM (
```

Comment:
Must this not be created in `pre`? Some functions already use it there.

Reply:
I still think this is true. I will move it over to `procrastinate/sql/migrations/03.00.00_01_pre_cancel_notification.sql`